rpcserver.go

```diff
@@ -1369,6 +1424,38 @@ func (s *rpcServer) validateOrder(o order.Order, acct *account.Account,
 		}
 	}
 
+	// Taproot channels require lnd 0.17.0 or higher.
+	if o.Details().ChannelType == order.ChannelTypeSimpleTaproot {
+		taprootNotSupportedErr := fmt.Errorf("connected lnd version "+
+			"%v does not support taproot channels, need at least "+
+			"v%d.%d.%d with --protocol.simple-taproot-chans "+
+			"turned on", s.server.lndServices.Version.Version,
+			taprootChannelsVersion.AppMajor,
+			taprootChannelsVersion.AppMinor,
+			taprootChannelsVersion.AppPatch)
+
+		// First check the lnd version itself.
```

Comment:
I think we can check this on start up, the cache the response so we don't need to hit version each time for an order submission.

Reply:
We do exactly that. The variable `s.server.lndServices.Version` contains the cached version and `lndclient.AssertVersionCompatible()` just does the comparison of the two versions.